### PR TITLE
allow restriction of counts in callback of waveforms

### PR DESCRIPTION
**Pull request by aschaelicke**
as discussed in issue #72

### CODE REVIEW

**Comment by newville** [comment]
@aschaelicke This seems like a good idea to me.   I'd prefer the default value for `count` in `PV.__init__()` to be `None` instead of -1, and then use 
```python
     if self._args['count'] is None:
         self._args['count'] = count
     else:
         self._args['count'] = min(count, self._args['count']
```

It's not completely obvious to me what NELM should be in this case -- the "native length" on the server, or the maximum count for the subscription.  

It looks like several tests with waveforms fail with this change.  These might be cases where count=0 and/or 1, which can be tricky.  It might also be that there are assumptions about NORD and NELM that no longer hold when the subscription is restricted.  Anyway, these failures will have to be resolved, and ideally more tests added for this feature.

File: epics/pv.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ class PV(object):
                'upper_warning_limit', 'upper_ctrl_limit', 'lower_ctrl_limit')
 
     def __init__(self, pvname, callback=None, form='time',
-                 verbose=False, auto_monitor=None,
+                 verbose=False, auto_monitor=None, count= -1,
```
Comment (tacaswell) [inline epics/pv.py]: I would prefer `None` as a sentinal, not -1

**Comment by TBirkeHZB** [comment]
We changed handling of optional count argument. 
Works properly now and passes all unittests - including the unittests we added for the new feature.

**Comment by newville** [comment]
@TBirkeHZB  Thanks for the update and the test.  I like the separate "maxcount" too.

This looks good to me -- any objections to merging?

**Comment by TBirkeHZB** [comment]
Not at all... And it would save Andreas (@aschaelicke) from applying his patch to every new version of pyepics in our local installations ;-) Thanks a lot!

**Comment by newville** [comment]
OK, merged.  Thanks!

**Pull request merged**